src/services/PostService.ts

```diff
@@ -165,7 +165,10 @@ export class PostService {
       const post = await postRepository.getPostById(params.id);
       if (!post) throw new NotFoundError('Post not found!');
       const userRepository = Repositories.user(transactionalEntityManager);
-      return await userRepository.isSavedPost(user, post);
+      const newUser = await userRepository.getSavedPostsByUserId(user.id);
+      if (!newUser) throw new NotFoundError('User not found!');
+      const isSaved = (p: PostModel) => p.id == post.id;
+      return newUser.saved.some(isSaved);
```

Comment:
why is this function called `isSavedPost` and then it gets saved **posts** and you check if at least one is saved? there are a lot of inconsistencies. if this were actually checking multiple posts I'd expect to do `saved.all(isSaved)` to check if everything is saved.